lib/index.js

```diff
@@ -300,73 +299,69 @@ internals.stop = async (server) => {
 
-    if (parsedArgs.mode) {
-        Hoek.assert(parsedArgs.mode === 'create' || parsedArgs.mode === 'alter', new ctx.DisplayError(Bossy.usage(definition)));
-    }
+    // Determine realm's knex and migrationsDir
 
     const { realmsWithMigrationsDir } = internals.state(srv.realm).collector;
-
-    const realmsWithMigrationsInRoot = Array.from(realmsWithMigrationsDir).reduce((collector, realm) => {
+    const realmsWithMigrationsInRoot = Array.from(realmsWithMigrationsDir).filter((realm) => {
 
         const { migrationsDir } = internals.state(realm);
-        if (internals.isChildPath(migrationsDir, rootDir)) {
-            collector.push(realm);
-        }
 
-        return collector;
-    }, []);
+        return internals.isChildPath(migrationsDir, rootDir);
+    });
 
-    Hoek.assert(realmsWithMigrationsInRoot.length > 0, new ctx.DisplayError(`At least 1 migrations directory must be present inside ${rootDir}`));
-    Hoek.assert(realmsWithMigrationsInRoot.length === 1, new ctx.DisplayError(`Only 1 migrations directory supported per project at this time. Please remove one of [${realmsWithMigrationsInRoot.map((realm) => realm.plugins.schwifty.migrationsDir)}]`));
+    Hoek.assert(realmsWithMigrationsInRoot.length > 0, new DisplayError(`At least one migrations directory must be present inside ${rootDir}.`));
+    Hoek.assert(realmsWithMigrationsInRoot.length === 1, new DisplayError(`Only one migrations directory supported per project at this time. Please remove one of ${realmsWithMigrationsInRoot.map((realm) => internals.state(realm).migrationsDir).map((dir) => Path.relative(options.cwd, dir)).join(', ')}.`));
 
-    const [migrationRealm] = realmsWithMigrationsInRoot;
+    const { knex, migrationsDir } = internals.state(realmsWithMigrationsInRoot[0]);
 
-    const serverModels = srv.models();
-    const allModels = Object.keys(serverModels).map((modelName) => serverModels[modelName]);
-    const realmKnex = internals.state(migrationRealm).knex;
-    const models = allModels.filter((model) => model.knex() === realmKnex);
-    const modelsArr = Object.keys(models).map((key) => models[key]);
+    const models = Object.values(srv.models())
+        .filter((Model) => Model.knex() === knex);
 
-    return await Util.promisify(SchwiftyMigrateDiff.genMigrationFile)({
-        models: modelsArr,
-        migrationsDir: parsedArgs.dir || internals.state(migrationRealm).migrationsDir,
-        knex: realmKnex,
+    return await Util.promisify(migrateDiff.genMigrationFile)({
+        knex,
+        models,
+        migrationsDir,
         mode: parsedArgs.mode,
         migrationName
     });
 };
 
-internals.partIsDotDot = (part) => part === '..';
 internals.isChildPath = (pathA, pathB) => { // A child of B
 
     const relParts = Path.relative(pathA, pathB).split(Path.sep);
-    return relParts.every(internals.partIsDotDot);
+    const partIsDotDot = (part) => part === '..';
```

Comment:
How do I decide if a func should be broken out into internals or left as the no-no, 'creating a func inside a func'?

Reply:
I just thought this was more readable, and it was no problem since `partIsDotDot` is only used once inside `isChildPath()`.  The function could also have been written directly as an argument to `some()`, but naming it makes this more readable.

Reply:
Also—it's not inherently bad to create a function inside of a function!  While there are cases where you might not want to, it's not something to always avoid.